internal/app/machined/pkg/runtime/v1alpha1/platform/cloudstack/metadata.go

```diff
@@ -0,0 +1,76 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package cloudstack
+
+import (
+	"context"
+	stderrors "errors"
+	"fmt"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/platform/errors"
+	"github.com/siderolabs/talos/pkg/download"
+)
+
+const (
+	// CloudstackMetadataEndpoint is the local Cloudstack endpoint.
+	CloudstackMetadataEndpoint = "http://data-server./latest/meta-data"
```

Comment:
Yes - as stated in the documentation linked:

_From inside the instance, the user-data is accessible via the virtual router, if the UserData service is enabled on the network offering._

_If you are using the DNS service of the virtual router, a special hostname called data-server. is provided, that will point to a valid user-data server._

_Otherwise you have to determine the virtual router address via other means, such as DHCP leases. Be careful to scan all routers if you have multiple networks attached to an instance, in case not all of them have the UserData service enabled._

_User-data is available from the URL http://data-server./latest/user-data and can be fetched via curl or other HTTP client._

At some point it might be an idea to add more sophisticated logic to obtain the address of the virtual router, but for most use cases using the virtual routers DNS would probably be the way to go.